tests/functional/step_defs/test_chart_src_with_report.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""Report and chart source submission
+
+Partners or redhat associates can publish their chart by submitting
+error-free chart in source format with the report.
+"""
+import pytest
+from pytest_bdd import (
+    given,
+    scenario,
+    then,
+    when,
+)
+
+from functional.utils.chart_certification import ChartCertificationE2ETestSingle
+
+@pytest.fixture
+def workflow_test():
+    test_name = 'Test Chart Source With Report'
+    test_chart = 'tests/data/vault-0.13.0.tgz'
+    test_report = 'tests/data/report.yaml'
+    workflow_test = ChartCertificationE2ETestSingle(test_name=test_name, test_chart=test_chart, test_report=test_report)
+    yield workflow_test
+    workflow_test.cleanup()
+
+
+@scenario('../features/report_and_chart_src.feature', "The partner hashicorp submits an error-free chart source with report for vault")
+def test_partner_chart_src_submission():
+    """The partner hashicorp submits an error-free chart source with report for vault."""
+
+
+@scenario('../features/report_and_chart_src.feature', "A redhat associate submits an error-free chart source with report for vault")
+def test_redhat_chart_src_submission():
+    """A redhat associate submits an error-free chart source with report for vault."""
+
+
+@given("hashicorp is a valid partner")
```

Comment:
For example, this ```@given``` function is repeated in multiple .py files. We should make these common too, by including the tests in the same bdd file.